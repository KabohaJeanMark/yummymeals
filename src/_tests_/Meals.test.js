import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import MealDetailComponent from '../containers/Meals';
import store from '../reducers';

describe('MealDetailComponent component tests', () => {
  it('should match the snapshot', () => {
    const tree = renderer.create(
      <BrowserRouter>
        <Provider store={store}>
          <MealDetailComponent />
        </Provider>
      </BrowserRouter>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should be defined', async () => {
    const component = render(
      <BrowserRouter>
        <Provider store={store}>
          <MealDetailComponent />
        </Provider>
      </BrowserRouter>,
    );
    expect(component).toBeDefined();
  });
});
