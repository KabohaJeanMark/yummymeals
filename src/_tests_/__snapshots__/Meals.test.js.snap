// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MealDetailComponent component tests should match the snapshot 1`] = `
<div
  className="container-fluid"
>
  <p
    className="mt-2"
  >
    Please select a food category from the dropdown :)
  </p>
  <select
    aria-label="Default select example"
    className="form-select m-2"
    id="category"
    name="category"
    onChange={[Function]}
    placeholder="Food category"
  >
    <option
      value="Breakfast"
    >
      Breakfast
    </option>
    <option
      value="Dessert"
    >
      Dessert
    </option>
    <option
      value="Vegan"
    >
      Vegan
    </option>
    <option
      value="Chicken"
    >
      Chicken
    </option>
    <option
      value="Side"
    >
      Side
    </option>
    <option
      value="Pasta"
    >
      Pasta
    </option>
  </select>
  <div
    className="mealsContainer"
  />
</div>
`;
